### DISCUSSION TITLE
Pi-hole-for-Android v1.7

### DISCUSSION THREAD

**Discussion by DesktopECHO** [DesktopECHO/Pi-hole-for-Android#22] [Announcements]
- Update p4a-install
- Update rc.local init scripts
- Added ntp for time sync on startup
- Better handling of network discovery when device DNS is pointing at local Pi-hole instance
- Note: Enable Debug Mode in **Settings** to see console output below:

![Screenshot_20221006-095149](https://user-images.githubusercontent.com/33142753/194333451-d9b1f03d-99c9-4ac6-8e64-cc7469165960.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/DesktopECHO/Pi-hole-for-Android/releases/tag/v1.7'>Pi-hole-for-Android v1.7</a>.</em>

**Comment by ghost** [comment]
[Tested on TVBox RK3318](https://ibb.co/F30rRPQ)
No problems found,

**Comment by n0w33d** [comment]
Do you recommend reinstalling this tgz on our android devices in LinuxDeploy, or does running `p4a-install` in console update to this release automatically?

**Comment by DesktopECHO** [comment]
If you are on v1.6 or older, you'll need to install the 1.7 image.  This is due to changes upstream that are outside my control.

After that, running `p4a-install` from v1.7 and newer will keep you up to date.

**Comment by n0w33d** [comment]
I had DHCP issues from version 1.6 that forced me to only use DNS via p4a and DHCP pool delegation from the router; those issues went away with this new version and I can now use both  DNS & DHCP services on the Android device instead of splitting those between my router and Android device. Thank you.

**Comment by n0w33d** [comment]
Disregard. My network was not using the pihole DHCP, and is indeed not working after 24 hours (which was the DHCP lease time that was setup on the router).

I will try to investigate further and maybe post on the issues page.

**Comment by DesktopECHO** [comment]
It should work... Anything interesting in /var/log/pihole-FTL.log ?  

![image](https://user-images.githubusercontent.com/33142753/196091652-2bf7a0f5-b6bf-422b-8de7-7a0473b6d0de.png)

**Reply by n0w33d** [reply]
Hello, thanks for replying.

I see no erroneous output in the FTL log, but I will try another attempt and get back to you here soon.

My DHCP leases populate the table on the FTL, but once I disable the DHCP server on my router, within 24 hours the Pihole host device needs a static IP assignment and default route assignment via an Android terminal emulator, and nothing gets routed to the correct device on the network regardless.

 I think this is caused by a possible Android LineageOS/AOSP driver issue due to me using a [Belkin powered USB-C to Ethernet adapter](https://www.belkin.com/us/business/audio-video-adapters/usb-c-to-ethernet-charge-adapter/p/p-inc001/); I have no idea what's wrong yet.

**Comment by DesktopECHO** [comment]
Interesting, never came across one of these before that had Android support out of the box.
What kind of device is it plugged into?

You will definitely need to go into settings and assign a static IP address before enabling DHCP... but now I am seeing online that 'some' devices don't let you set a static ip.  Is that your situation as well?  If so, a LineageOS ROM may expose the static IP settings you need.

Worst case scenario, you have to leave DCHP enabled on your router, but the router's DHCP pool should only be 1 IP, and that IP should be a static reservation for your Pi-hole.  

That way, a DHCP device on your network will seek an IP first from your router, but the router will tell them "nope, sorry my pool is full" and the device will get an IP from the Pi-hole DHCP server instead.

**Reply by n0w33d** [reply]
My device is plugged into a [Belkin USB-C to Ethernet](https://www.belkin.com/us/business/audio-video-adapters/usb-c-to-ethernet-charge-adapter/p/p-inc001/) charging adapter.

I set a static IP as well as duplicated the IP rules that are auto configured if I used DHCP service from my router, I also set a default route via the routers default gateway address, all directly inputted from the device's local terminal emulator as root. Nothing, no network is accessible. The LineageOS rom does not add any solution so far.

Regarding your worst-case scenario: I have indeed set this up and the router defies logic and provides DHCP assignments beyond the pool limits (please refer to screenshot).
![Screenshot Cens](https://user-images.githubusercontent.com/106119998/196217990-a32455b8-51ad-4792-92af-b440b46681b1.png)

So, I still have no idea what is going on yet. Looking everywhere for a solution (initially i thought it was Unbound being dumb, or the resolv.conf existing, but the troubleshooting has come down to the adapter not talking to the router?).

I will keep investigating and thank you for the troubleshooting recommendations.

EDIT: I just took a gander at my own screenshot and noticed that had setup two static IP assignments, so I removed one and kept the PiHole device assigned. I will report back here if this is the solution.

EDIT 2: Report - this is not the solution. The Pihole is not filtering network traffic via DHCP delegation, only DNS.

**Reply by DesktopECHO** [reply]
- For `Starting Address of Local Pool`  use something other than `0` as that can be troublesome.  

- Seeing as how your router is `192.168.0.1` I'd recommend a starting address of `192.168.0.2`  --  Then reboot the router and see if multiple devices are still getting IPs.  

- There are some routers that have a _minimum_ pool size -- For example Eero devices require a minimum of 10 IP's for its DHCP pool.  To get around that particular issue, I had to create 10 'bogus' reservations for each IP in the pool.  After that, Pi-hole was able to hand out IP addresses. 

So I gather there is no way to set a static IP in the Android GUI on your device, is that correct?
What commands do you use in terminal to change your IP?

**As a test:** If you just run `ip address add 192.168.0.100 dev eth0` that should give you a second static IP for the interface, does it?

**Reply by n0w33d** [reply]
Thanks again for your reply.

I set the starting address field to "11" which is the same final octet for the static IP assignment of the Pihole device. Rebooted a couple devices and it seems to not show any new DHCP assignments in the router gateway page. 

I am now seeing a a spike in the activity graphs of the FTL admin page. I will continue to monitor and report back here again soon with an update.

My apologies for the troubleshooting I am distracting you that is unrelated to your project here.

**Reply by n0w33d** [reply]
Looks like your suggestion fixed my issue.

Thank you so much!

**Comment by Throyiii** [comment]
![Screenshot_20221018_152630](https://user-images.githubusercontent.com/100574683/196364910-f1b04244-810d-4a88-af9e-a5f051391069.jpg)
![Screenshot_20221018_152428](https://user-images.githubusercontent.com/100574683/196364957-a1c47f20-ecf1-4fa4-b1c6-b0a05f5bef6e.jpg)


Thanks

**Reply by DesktopECHO** [reply]
Thanks for the success report!
